proof/records.go

```diff
@@ -12,37 +12,37 @@ import (
 
 const (
 	VersionType          tlv.Type = 0
-	PrevOutType          tlv.Type = 1
-	BlockHeaderType      tlv.Type = 2
-	AnchorTxType         tlv.Type = 3
-	TxMerkleProofType    tlv.Type = 4
-	AssetLeafType        tlv.Type = 5
-	InclusionProofType   tlv.Type = 6
-	ExclusionProofsType  tlv.Type = 7
-	SplitRootProofType   tlv.Type = 8
-	MetaRevealType       tlv.Type = 9
-	AdditionalInputsType tlv.Type = 10
-	ChallengeWitnessType tlv.Type = 11
-	BlockHeightType      tlv.Type = 12
-	GenesisRevealType    tlv.Type = 13
-	GroupKeyRevealType   tlv.Type = 14
+	PrevOutType          tlv.Type = 2
+	BlockHeaderType      tlv.Type = 4
+	AnchorTxType         tlv.Type = 6
+	TxMerkleProofType    tlv.Type = 8
+	AssetLeafType        tlv.Type = 10
+	InclusionProofType   tlv.Type = 12
```

Comment:
One thing we even odd that we could do here is actually assign in pairs. We do this for feature bits in LN so we can flip them to required if needed. 

In this case, we would have `ExclusionProofsType` actually be `15`. 

Not sure if this is useful though?

Reply:
Ah, I see. Yes, that could be nice so we could turn things on/off when it comes to being required. Happy to do the change if you think we should. But then I'd say we should update the address records as well.

Reply:
Don't feel super strongly about this, as in the future if we wish we can use a new version to redo all the types. So far we don't have a concrete need for this as is though.

Reply:
Yeah, me neither... So I guess let's keep it the way they are now?